internal/services/cdn/cdn_endpoint_custom_domain_resource.go

```diff
@@ -159,3 +351,178 @@ func resourceArmCdnEndpointCustomDomainDelete(d *pluginsdk.ResourceData, meta in
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	raw := input[0].(map[string]interface{})
+	output := &cdn.ManagedHTTPSParameters{
+		CertificateSourceParameters: &cdn.CertificateSourceParameters{
+			OdataType:       utils.String("#Microsoft.Azure.Cdn.Models.CdnCertificateSourceParameters"),
+			CertificateType: cdn.CertificateType(raw["certificate_type"].(string)),
+		},
+		CertificateSource: cdn.CertificateSourceCdn,
+		ProtocolType:      cdn.ProtocolType(raw["protocol_type"].(string)),
+		MinimumTLSVersion: cdn.MinimumTLSVersionNone,
+	}
+
+	return output
+}
+
+func expandArmCdnEndpointCustomDomainUserManagedHttpsSettings(input []interface{}) (cdn.BasicCustomDomainHTTPSParameters, error) {
+	if len(input) == 0 || input[0] == nil {
+		return nil, nil
+	}
+
+	raw := input[0].(map[string]interface{})
+
+	keyVaultId, err := keyvaultParse.VaultID(raw["key_vault_id"].(string))
+	if err != nil {
+		return nil, err
+	}
+
+	output := &cdn.UserManagedHTTPSParameters{
+		CertificateSourceParameters: &cdn.KeyVaultCertificateSourceParameters{
+			OdataType:         utils.String("#Microsoft.Azure.Cdn.Models.KeyVaultCertificateSourceParameters"),
+			SubscriptionID:    &keyVaultId.SubscriptionId,
+			ResourceGroupName: &keyVaultId.ResourceGroup,
+			VaultName:         &keyVaultId.Name,
+			SecretName:        utils.String(raw["secret_name"].(string)),
+			SecretVersion:     utils.String(raw["secret_version"].(string)),
+			UpdateRule:        utils.String("NoAction"),
+			DeleteRule:        utils.String("NoAction"),
+		},
+		CertificateSource: cdn.CertificateSourceAzureKeyVault,
+		ProtocolType:      cdn.ProtocolTypeServerNameIndication,
+		MinimumTLSVersion: cdn.MinimumTLSVersionNone,
+	}
+
+	return output, nil
+}
+
+func flattenArmCdnEndpointCustomDomainCdnManagedHttpsSettings(input cdn.ManagedHTTPSParameters) []interface{} {
+	certificateType := ""
+	if params := input.CertificateSourceParameters; params != nil {
+		certificateType = string(params.CertificateType)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"certificate_type": certificateType,
+			"protocol_type":    string(input.ProtocolType),
+		},
+	}
+}
+
+func flattenArmCdnEndpointCustomDomainUserManagedHttpsSettings(input cdn.UserManagedHTTPSParameters) []interface{} {
+	var (
+		subscriptionId    string
+		resourceGroupName string
+		vaultName         string
+		secretName        string
+		secretVersion     string
+	)
+	if params := input.CertificateSourceParameters; params != nil {
+		if params.SubscriptionID != nil {
+			subscriptionId = *params.SubscriptionID
+		}
+		if params.ResourceGroupName != nil {
+			resourceGroupName = *params.ResourceGroupName
+		}
+		if params.VaultName != nil {
+			vaultName = *params.VaultName
+		}
+		if params.SecretName != nil {
+			secretName = *params.SecretName
+		}
+		if params.SecretVersion != nil {
+			secretVersion = *params.SecretVersion
+		}
+	}
+
+	return []interface{}{
```

Comment:
(as above) we should raise an error here if Subscription ID / Resource Group / Vault aren't all returned?